src/loaders/lottie/tvgLottieBuilder.cpp

```diff
@@ -1000,11 +1000,14 @@ static void _updatePrecomp(LottieLayer* precomp, float frameNo)
         clipper->appendRect(0, 0, static_cast<float>(precomp->w), static_cast<float>(precomp->h));
         clipper->transform(precomp->cache.matrix);
 
-        //TODO: remove the intermediate scene....
-        auto cscene = Scene::gen();
-        cscene->composite(cast(clipper), CompositeMethod::ClipPath);
-        cscene->push(cast(precomp->scene));
-        precomp->scene = cscene.release();
+        if (precomp->scene->composite(nullptr) != CompositeMethod::None) {
```

Comment:
We assume the use of the new viewport API here.

Reply:
but the clipper is transformed, it may not be a rectangle

Reply:
The primary purpose of this viewport optimization is for this specific application, not others. If it's not applicable here, then it might be unnecessary. Perhaps we should internally consider whether the final transformed viewport is axis-aligned. Otherwise, this idea is wrong approach from the very-beginning.